Seasonal/StaffBirthdays/Nulgath/TempleDelveMerge.cs

```diff
@@ -0,0 +1,133 @@
+        Core.SetOptions();
+
+        BuyAllMerge();
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge(string buyOnlyThis = null, mergeOptionsEnum? buyMode = null)
+    {
+        if (!Core.isSeasonalMapActive("templedelve"))
+            return;
+
+        TD.Storyline();
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("templedelve", 2232, findIngredients, buyOnlyThis, buyMode: buyMode);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Doomed Extract":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.RegisterQuests(9090);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.EquipClass(ClassType.Farm);
+                        Core.HuntMonster("templedelve", "Delirious Elemental", "Elemental Study", 6, log: false);
+                        Core.HuntMonster("templedelve", "Infested Nation", "Infestation Study", 6, log: false);
+                        Core.EquipClass(ClassType.Solo);
+                        Core.HuntMonster("templedelve", "Doomed Fiend", "Fiend Worm", log: false);
+                        Bot.Wait.ForPickup(req.Name);
+                    }
+                    Core.CancelRegisteredQuests();
+                    break;
+
+                case "Tainted Gem":
+                    Nation.SwindleBulk(quant);
+                    break;
+
+                case "Dark Crystal Shard":
+                    Nation.FarmDarkCrystalShard(quant);
+                    break;
+
+                case "Diamond of Nulgath":
+                    Nation.FarmDiamondofNulgath(quant);
+                    break;
+
+                case "Nation Ritualist":
+                case "Void Nation Ritualist":
+                    Core.FarmingLogger(req.Name, quant);
+                    Core.EquipClass(ClassType.Solo);
+                    while (!Bot.ShouldExit && !Core.CheckInventory(req.Name, quant))
+                    {
+                        Core.HuntMonster("templedelve", "Doomed Fiend", req.Name, isTemp: false, log: false);
```

Comment:
No need to put this inside a while loop, entire case could just be a swap class and hunt monster